autobahn/wamp/component.py

```diff
@@ -547,12 +549,13 @@ def on_disconnect(session, was_clean):
                         was_clean=was_clean,
                     )
                     if not txaio.is_called(done):
-                        if was_clean:
-                            # eg the session has left the realm, and the transport was properly
-                            # shut down. successfully finish the connection
-                            txaio.resolve(done, None)
-                        else:
-                            txaio.reject(done, RuntimeError('transport closed uncleanly'))
```

Comment:
the deferred should be rejected in this case

Reply:
Yeah it does seem like that should be true -- but I changed this because it makes the shutdown wrong if the closing handshake timeout is hit. What happens then is that the component is trying to shut down, but then gets an error while closing -- so we either need a custom exception here that we catch in here: https://github.com/crossbario/autobahn-python/blob/master/autobahn/twisted/component.py#L372 (and then stop trying to re-connect) .. "or something".

Basically I don't think we care at this point if the transport closed cleanly or uncleanly (as far as Component behavior) and just need to log it...